lib/libzfs/libzfs_sendrecv.c

```diff
@@ -3656,7 +3777,57 @@ zfs_receive_one(libzfs_handle_t *hdl, int infd, const char *tosnap,
 		nvlist_free(prop_errors);
 
 	if (newprops)
-		nvlist_free(props);
+		nvlist_free(rcvprops);
+
+	if (origprops != NULL)
+		nvlist_free(origprops);
+
+	return (err);
+}
+
+/*
+ * Check properties we were asked to override (both -o|-x)
+ */
+static int
+zfs_receive_checkprops(libzfs_handle_t *hdl, nvlist_t *props,
+    const char *errbuf)
+{
+	nvpair_t *nvp;
+	int err = 0;
+	const char *name;
```

Comment:
for consistency, you should either define prop and name variables outside the while or inside